### PR TITLE
Fix headers, let request service handle them

### PR DESCRIPTION
**Pull request by ffiebig**
Changed the way custom headers for Patcomercio are set. They used to be public when they shouldn't. 

Now you can call RequestService with custom headers and it is done that way inside classes that need it (Patcomercio Inscription class)

Removed alternative of adding headers to options, that's not something the user should take care of...

These are breaking changes on something people shouldn't use, anyway, we should bump it to the next version

### CODE REVIEW

**Comment by gonzunigad** [comment]
It is something completely needed or it is just an improvement? 
As you said it could be approved for the next 3.X release.

**Comment by ffiebig** [comment]
I'd say it's needed, just because the Options class shouldn't be asking for headers

**Comment by gonzunigad** [comment]
As this was just released on 2.5.X, maybe we can merge it as a bugfix. Does this change the Options constructor signature for Webpay Plus, OneClick and Tx Completa also or just PatPass Comercio?

**Comment by ffiebig** [comment]
It's the same class for all of them

**Comment by ffiebig** [comment]
So yeah, it changes the signature of all those products, the thing is, it could lead to errors, there's no point in letting the user change the headers if we know them beforehand

**Comment by gonzunigad** [comment]
In that case it can not be a bugfix because OneClick Mall and Tx Completa has been available for months now and it is used by many commerces. 

I would not change this until 3.X. (we can remove soap and other breaking changes on the same release)

**Comment by ffiebig** [comment]
I'll make another PR to add the repeated code that we need to make Webpay Plus Deferred work instead

**Comment by ffiebig** [comment]
There, made a new PR

File: Transbank/Common/Options.cs [hunk 1]
```diff
@@ -46,12 +32,11 @@ public IIntegrationType IntegrationType
                 );
         }
 
-        public Options(string commerceCode, string apiKey, IIntegrationType integrationType, RequestServiceHeaders headers)
+        public Options(string commerceCode, string apiKey, IIntegrationType integrationType)
```
Comment (gonzunigad) [inline Transbank/Common/Options.cs:35]:
```suggestion
        public Options(string commerceCode, string apiKey, IIntegrationType integrationType, RequestServiceHeaders headers = null)
```

What if we keep that parameter but as optional?

Comment (gonzunigad) [reply Transbank/Common/Options.cs:35]: I think that would not be a breaking change.

**Comment by gonzunigad** [comment]
This PR has conflicts you should fix before merging :)

**Review by gonzunigad — APPROVED** [review]

**Pull request merged**